backend/src/sync/harvest/mod.rs

```diff
@@ -473,3 +481,53 @@ async fn upsert(
     let statement = db.prepare_cached(&*query).await?;
     Ok(db.query_one(&statement, &values).await?.get::<_, i64>(0))
 }
+
+
+/// Removes realms that reference a deleted series, event, or playlist,
+/// if the configuration allows it.
+///
+/// If realm deletion is enabled for the given type, the function:
+/// 1. Finds all realms whose names are derived from the deleted series/event/playlist.
+/// 2. Ensures that those realms have no child realms.
+/// 3. If `eager = false`, ensures that the realm has **only one** block (the deleted one).
+/// 4. Deletes the qualifying realms from the database.
+async fn remove_realms(
+    db: &deadpool_postgres::Transaction<'_>,
+    config: &Config,
+    item: &HarvestItem,
+) -> Result<(), tokio_postgres::Error> {
+    let Some(props) = item.deleted_props(config) else {
+        return Ok(());
+    };
+
+    let block_count_condition = if props.eager {
+        String::new()
+    } else {
+        "and (select count(*) from blocks b2 where b2.realm = r.id) = 1".to_owned()
+    };
```

Comment:
```suggestion
    let block_count_condition = if props.eager {
        ""
    } else {
        "and (select count(*) from blocks b2 where b2.realm = r.id) = 1"
    };
```

No need for string allocation here.